circuit-lib/circuit-lib.js/src/elgamal-babyjubjub/elgamal.ts

```diff
@@ -0,0 +1,171 @@
+import { AffinePoint } from "@noble/curves/abstract/curve";
+import { ExtPointType } from "@noble/curves/abstract/edwards";
+import { babyJub as CURVE } from "./babyjubjubConfig";
+import { encode } from "./pointCodec";
+import {
+  genPubKey,
+  genPrivKey,
+  formatPrivKeyForBabyJub,
+  genRandomSalt,
+} from 'maci-crypto'
+
+const ff = require("ffjavascript");
+
+type BabyJubAffinePoint = AffinePoint<bigint>;
+type BabyJubExtPoint = ExtPointType;
+type SecretKey = bigint;
+
+type PublicKey = BabyJubExtPoint;
+
+interface Keypair {
+  secretKey: SecretKey;
+  publicKey: BabyJubExtPoint;
+}
+
+const babyJub = CURVE.ExtendedPoint;
+
+const genKeypair = (): Keypair => {
+  const secretKey = genPrivKey() as bigint;
+  const publicKey = genPubKey(secretKey);
+  const publicKey_ext = ElGamalUtils.coordinatesToExtPoint<BigInt>(publicKey[0], publicKey[1]);
+
+  const Keypair: Keypair = { secretKey, publicKey: publicKey_ext }
+
+  return Keypair
+}
+/**
+ * @returns A random extended point on the Baby Jubjub curve.
+ */
+function genRandomPoint(): BabyJubExtPoint {
+  const salt = genRandomSalt();
+  const randomPoint = genPubKey(salt);
+  const randomPoint_ext = ElGamalUtils.coordinatesToExtPoint<BigInt>(randomPoint[0], randomPoint[1]);
+  return randomPoint_ext;
+}
+
+/**
+ * Encrypts a plaintext such that only the owner of the specified public key
+ * may decrypt it.
+ * @param publicKey The recepient's public key
+ * @param plaintext A 32-bit bigint -> to be encoded and then encrypted
+ * @param randomVal A random value used along with the secret key to generate the ciphertext (optional)
+ */
+function encrypt(publicKey: BabyJubExtPoint, plaintext: bigint, randomVal?: bigint) {
```

Comment:
Is there test coverage for this function?

Reply:
done!